### PR TITLE
Fix and cleanup tests, update dependencies

### PR DESCRIPTION
**Pull request by SUPERCILEX**
@ciarand here you go!

### CODE REVIEW

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java [hunk 1]
```diff
@@ -16,18 +16,21 @@
...
 import org.robolectric.annotation.Config;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java]: nit: preferred import style is static imports (sorted alphabetically), a blank line, then everything else (sorted alphabetically)

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java [hunk 2]
```diff
@@ -16,18 +16,21 @@
...
-import java.util.List;
+
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java]: Please leave the `manifest = Config.NONE` part in - it's necessary for these tests to be able to run both inside and outside google

Comment (SUPERCILEX) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java]: Done.

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java (hunk 1 above)
Comment (SUPERCILEX) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java]: And done.

File: .travis.yml [hunk 3]
```diff
@@ -1,4 +1,3 @@
-# https://docs.travis-ci.com/user/languages/android
```
Comment (ciarand) [inline .travis.yml]: let's leave this for convenience

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java [hunk 4]
```diff
@@ -19,15 +19,17 @@
 import static org.junit.Assert.assertEquals;
 
 import android.text.TextUtils;
-import java.util.Arrays;
-import java.util.List;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/ConstraintTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/DefaultJobValidatorTest.java [hunk 5]
```diff
@@ -21,25 +21,27 @@
...
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/DefaultJobValidatorTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/DefaultJobValidatorTest.java [hunk 6]
```diff
@@ -56,25 +58,25 @@ public void testValidate_retryStrategy() throws Exception {
...
-            Arrays.asList("Maximum backoff must be greater than 300s (5 minutes)"));
+            Collections.singletonList("Maximum backoff must be greater than 300s (5 minutes)"));
         testCases.put(
             new RetryStrategy(RetryStrategy.RETRY_POLICY_LINEAR, 301, 300),
-            Arrays.asList("Maximum backoff must be greater than or equal to initial backoff"));
+            Collections.singletonList("Maximum backoff must be greater than or equal to initial backoff"));
         testCases.put(
             new RetryStrategy(RetryStrategy.RETRY_POLICY_EXPONENTIAL, 301, 300),
-            Arrays.asList("Maximum backoff must be greater than or equal to initial backoff"));
+            Collections.singletonList("Maximum backoff must be greater than or equal to initial backoff"));
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/DefaultJobValidatorTest.java]: These lines are too long (> 100) with the full method name. Can you static import Collections.singletonList?

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/ExternalReceiverTest.java [hunk 7]
```diff
@@ -34,22 +34,24 @@
...
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.RuntimeEnvironment;
 import org.robolectric.annotation.Config;
 
-@SuppressWarnings("WrongConstant")
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/ExternalReceiverTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/FirebaseJobDispatcherTest.java [hunk 8]
```diff
@@ -36,18 +35,21 @@
...
 import org.mockito.invocation.InvocationOnMock;
 import org.mockito.stubbing.Answer;
-import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
+import java.util.Arrays;
+import java.util.List;
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/FirebaseJobDispatcherTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayCallbackExtractorTest.java [hunk 9]
```diff
@@ -24,18 +24,20 @@
 import android.os.Parcel;
 import android.os.Parcelable;
 import android.os.RemoteException;
+
 import com.google.android.gms.gcm.INetworkTaskCallback;
 import com.google.android.gms.gcm.PendingCallback;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayCallbackExtractorTest.java]: nit: unnecessary spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayDriverTest.java [hunk 10]
```diff
@@ -34,25 +34,25 @@
 import android.os.Parcelable;
 import android.support.annotation.NonNull;
 import android.text.TextUtils;
-import java.util.Arrays;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayDriverTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayJobWriterTest.java [hunk 11]
```diff
@@ -16,24 +16,27 @@
...
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayJobWriterTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobCoderTest.java [hunk 12]
```diff
@@ -20,19 +20,19 @@
 import static org.junit.Assert.assertNull;
 
 import android.os.Bundle;
+
 import com.firebase.jobdispatcher.Job.Builder;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobCoderTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobInvocationTest.java [hunk 13]
```diff
@@ -20,13 +20,14 @@
 import static org.junit.Assert.assertTrue;
 
 import android.support.annotation.NonNull;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobInvocationTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobServiceTest.java [hunk 14]
```diff
@@ -27,21 +27,23 @@
...
 import org.junit.runner.RunWith;
-import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
-public class JobServiceTest {
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobServiceTest.java]: nit: import order / spacing

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/ValidationEnforcerTest.java [hunk 15]
```diff
@@ -23,32 +23,37 @@
...
 import org.mockito.MockitoAnnotations;
-import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class, manifest = Config.NONE, sdk = 21)
+import java.util.Collections;
+import java.util.List;
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/ValidationEnforcerTest.java]: nit: import order / spacing

**Comment by SUPERCILEX** [comment]
Done.

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayJobWriterTest.java [hunk 16]
```diff
@@ -16,11 +16,13 @@
 
 package com.firebase.jobdispatcher;
 
+import android.os.Bundle;
+
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/GooglePlayJobWriterTest.java]: nit: a few too many blank lines here

File: jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobServiceTest.java [hunk 17]
```diff
@@ -16,6 +16,7 @@
 
 package com.firebase.jobdispatcher;
 
+
```
Comment (ciarand) [inline jobdispatcher/src/test/java/com/firebase/jobdispatcher/JobServiceTest.java]: nit: unnecessary blank line

**Comment by SUPERCILEX** [comment]
Done.

**Pull request merged**